implementations/rust/ockam/ockam_node/src/storage/database/migrations/20231006100000_create_database.sql

```diff
@@ -0,0 +1,187 @@
+
+CREATE TABLE policy
+(
+    resource   TEXT NOT NULL,
+    action     TEXT NOT NULL,
+    expression BLOB NOT NULL
+);
+
+CREATE TABLE named_identity
+(
+    identifier TEXT NOT NULL UNIQUE,
+    name       TEXT UNIQUE,
+    vault_name TEXT NOT NULL,
+    is_default INTEGER DEFAULT 0
+);
+
+CREATE TABLE node
+(
+    name                 TEXT PRIMARY KEY,
+    identifier           TEXT    NOT NULL,
+    verbosity            INTEGER NOT NULL,
+    is_default           INTEGER NOT NULL,
+    is_authority         INTEGER NOT NULL,
+    tcp_listener_address TEXT,
+    pid                  INTEGER
+);
+
+CREATE TABLE node_project
+(
+    node_name    TEXT PRIMARY KEY,
+    project_name TEXT NOT NULL
+);
+
+CREATE TABLE okta_config
+(
+    project_id      TEXT NOT NULL,
+    tenant_base_url TEXT NOT NULL,
+    client_id       TEXT NOT NULL,
+    certificate     TEXT NOT NULL,
+    attributes      TEXT
+);
+
+CREATE TABLE confluent_config
+(
+    project_id       TEXT NOT NULL,
+    bootstrap_server TEXT NOT NULL
+);
+
+CREATE TABLE vault
+(
+    name       TEXT PRIMARY KEY,
+    path       TEXT NOT NULL,
+    is_default INTEGER,
+    is_kms     INTEGER
+);
+
+CREATE TABLE signing_secret
+(
+    handle      BLOB PRIMARY KEY,
+    secret_type TEXT NOT NULL,
+    secret      BLOB NOT NULL
+);
+
+CREATE TABLE x25519_secret
+(
+    handle BLOB PRIMARY KEY,
+    secret BLOB NOT NULL
+);
+
+CREATE TABLE project
+(
+    project_id             TEXT PRIMARY KEY,
+    project_name           TEXT    NOT NULL,
+    is_default             INTEGER NOT NULL,
+    space_id               TEXT    NOT NULL,
+    space_name             TEXT    NOT NULL,
+    identifier             TEXT,
+    access_route           TEXT    NOT NULL,
+    authority_identity     TEXT,
+    authority_access_route TEXT,
+    version                TEXT,
+    running                INTEGER,
+    operation_id           TEXT
+);
+
+CREATE TABLE user_project
+(
+    user_email TEXT NOT NULL,
+    project_id TEXT NOT NULL
+);
+
+
+CREATE TABLE space
+(
+    space_id   TEXT PRIMARY KEY,
+    space_name TEXT    NOT NULL,
+    is_default INTEGER NOT NULL
+);
+
+CREATE TABLE user_space
+(
+    user_email TEXT NOT NULL,
+    space_id   TEXT NOT NULL
+);
+
+CREATE TABLE user_role
+(
+    user_id    INTEGER NOT NULL,
+    project_id TEXT    NOT NULL,
+    user_email TEXT    NOT NULL,
+    role       TEXT    NOT NULL,
+    scope      TEXT    NOT NULL
+);
+
+CREATE TABLE user
+(
+    email          TEXT PRIMARY KEY,
+    sub            TEXT    NOT NULL,
+    nickname       TEXT    NOT NULL,
+    name           TEXT    NOT NULL,
+    picture        TEXT    NOT NULL,
+    updated_at     TEXT    NOT NULL,
+    email_verified INTEGER NOT NULL,
+    is_default     INTEGER NOT NULL
+);
+
+CREATE TABLE credential
+(
+    name                  TEXT PRIMARY KEY,
+    issuer_identifier     TEXT NOT NULL,
+    issuer_change_history TEXT NOT NULL,
```

Comment:
That should probably be in the `identity` table

Reply:
Yes, I did more or less direct mapping but we could enforce a bit of normalization.